azkaban-common/src/main/java/azkaban/executor/ExecutionControllerUtils.java

```diff
@@ -249,7 +249,9 @@ public static void failEverything(final ExecutableFlow exFlow) {
       exFlow.setEndTime(time);
     }
 
-    exFlow.setStatus(Status.FAILED);
+    if (exFlow.getStatus() != Status.POD_FAILED) {
```

Comment:
Perhaps make this check more generic and switch the status to FAILED if the previous status is already a final one ?
Something like `if (!isStatusFinised(exFlow.getStatus())`